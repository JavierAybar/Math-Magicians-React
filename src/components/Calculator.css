* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
}

.calculator-container {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  height: 50vh;
}

.btn {
  width: 80px;
  height: 34px;
  border: 1px solid #d7d7d7;
  border-radius: 6px;
}

.btn.grey:hover {
  background: #cacaca;
}

.btn.orange {
  background: orange;
}

.btn.orange:hover {
  background: #fc8c19;
}

.btn.wide {
  width: 50%;
}

.btn-operator {
  background: rgb(250 182 58);
}

.buttons-container {
  display: flex;
  flex-direction: column;
  gap: 5px;
}

.btn-zero {
  grid-column: span 2;
  display: flex;
  width: 100%;
  padding-top: 6px;
  padding-left: 77px;
}

.display {
  grid-area: result;
  background: #545454;
  color: #fff;
  height: 44px;
  border-radius: 6px;
  text-align: end;
  padding-right: 10px;
  padding-top: 13px;
  width: 329px;
  margin-bottom: 4px;
}

.container-rows {
  display: flex;
  gap: 3px;
}

.container__quotes {
  width: 300px;
  height: 380px;
  margin: auto;
  border: #545454 2px solid;
  border-radius: 8px;
}

.quotes {
  height: 350px;
  width: 300px;
  list-style: none;
}

.title__quotes {
  text-align: center;
  color: #cacaca;
  background: #222;
  padding-bottom: 5px;
}

.loading {
  text-align: center;
  font-weight: 600;
  font-size: 2rem;
}

.error {
  text-align: center;
  font-size: 2.3rem;
  color: red;
}
