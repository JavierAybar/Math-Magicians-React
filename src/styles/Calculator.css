* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
}

.calculator-container {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  width: 380px;
  height: 310px;
  background-color: #727272;
  border-radius: 8px;
}

.btn {
  width: 80px;
  height: 34px;
  border: 1px solid #d7d7d7;
  border-radius: 6px;
}

.btn.grey:hover {
  background: #cacaca;
}

.btn.orange {
  background: orange;
}

.btn.orange:hover {
  background: #fc8c19;
}

.btn.wide {
  width: 50%;
}

.btn-operator {
  background: rgb(250 182 58);
}

.buttons-container {
  display: flex;
  flex-direction: column;
  gap: 5px;
}

.btn-zero {
  grid-column: span 2;
  display: flex;
  width: 100%;
  padding-top: 6px;
  padding-left: 77px;
}

.display {
  grid-area: result;
  background: #545454;
  color: #fff;
  height: 44px;
  border-radius: 6px;
  text-align: end;
  padding-right: 10px;
  padding-top: 13px;
  width: 329px;
  margin-bottom: 4px;
}

.container-rows {
  display: flex;
  gap: 3px;
}

.container__quotes {
  width: 300px;
  height: 380px;
  margin: auto;
  border-radius: 8px;
  margin-top: 170px;
}

.quotes {
  height: 350px;
  width: 300px;
  border-radius: 0 0 6px 6px;
  background-color: #ededed;
  box-shadow: 2px 2px 4px 2px rgba(120, 120, 120, 0.3);
  list-style: none;
}

.title__quotes {
  text-align: center;
  color: #cacaca;
  background: #222;
  padding: 8px;
  padding-bottom: 5px;
  border-radius: 10px 10px 0 0;
  box-shadow: 2px 0 4px 2px rgba(120, 120, 120, 0.3);
}

.loading {
  font-family: math, sans-serif;
  text-align: center;
  font-weight: 600;
  font-size: 2rem;
  margin-top: 300px;
}

.error {
  text-align: center;
  font-size: 2.3rem;
  margin-top: 300px;
  color: red;
}

.section-quotes {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 100%;
  padding: 20px;
}

.section-quotes li {
  font-size: 1.3rem;
  font-family: math, sans-serif;
}
