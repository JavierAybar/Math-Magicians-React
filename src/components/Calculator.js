import './Calculator.css';

const Calculator = () => (
  <div className="calculator-container">
    <ChildContent />
  </div>
);

const ChildContent = () => (
  <div className="calculator">
    <p className="result">0</p>
    <section className="number">
      <button className="btn" type="button">AC</button>
      <button className="btn" type="button">+/-</button>
      <button className="btn" type="button">%</button>
      <button className="btn" type="button">7</button>
      <button className="btn" type="button">8</button>
      <button className="btn" type="button">9</button>
      <button className="btn" type="button">4</button>
      <button className="btn" type="button">5</button>
      <button className="btn" type="button">6</button>
      <button className="btn" type="button">1</button>
      <button className="btn" type="button">2</button>
      <button className="btn" type="button">3</button>
      <button className="btn btn-zero" type="button">0</button>
      <button className="btn" type="button">.</button>
    </section>
    <section className="operator">
      <button className="btn btn-operator" type="button">รท</button>
      <button className="btn btn-operator" type="button">X</button>
      <button className="btn btn-operator" type="button">-</button>
      <button className="btn btn-operator" type="button">+</button>
      <button className="btn btn-operator" type="button">=</button>
    </section>
  </div>
);
export default Calculator;
