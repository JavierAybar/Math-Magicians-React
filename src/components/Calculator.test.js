import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Calculator from './Calculator';

describe('Calculator Component', () => {
  test('renders the Calculator component', () => {
    const { container } = render(<Calculator />);
    expect(container).toMatchSnapshot();
  });
});
