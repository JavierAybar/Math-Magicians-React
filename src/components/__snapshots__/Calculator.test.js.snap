// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Component renders the Calculator component 1`] = `
<div>
  <div
    class="calculator-container"
  >
    <div
      class="display"
    >
      <div>
        0
      </div>
    </div>
    <section
      class="buttons-container"
    >
      <div
        class="container-rows"
      >
        <button
          class="btn  grey"
          type="button"
        >
          AC
        </button>
        <button
          class="btn  grey"
          type="button"
        >
          +/-
        </button>
        <button
          class="btn  grey"
          type="button"
        >
          %
        </button>
        <button
          class="btn orange"
          type="button"
        >
          รท
        </button>
      </div>
      <div
        class="container-rows"
      >
        <button
          class="btn  grey"
          type="button"
        >
          7
        </button>
        <button
          class="btn  grey"
          type="button"
        >
          8
        </button>
        <button
          class="btn  grey"
          type="button"
        >
          9
        </button>
        <button
          class="btn orange"
          type="button"
        >
          x
        </button>
      </div>
      <div
        class="container-rows"
      >
        <button
          class="btn  grey"
          type="button"
        >
          4
        </button>
        <button
          class="btn  grey"
          type="button"
        >
          5
        </button>
        <button
          class="btn  grey"
          type="button"
        >
          6
        </button>
        <button
          class="btn orange"
          type="button"
        >
          -
        </button>
      </div>
      <div
        class="container-rows"
      >
        <button
          class="btn  grey"
          type="button"
        >
          1
        </button>
        <button
          class="btn  grey"
          type="button"
        >
          2
        </button>
        <button
          class="btn  grey"
          type="button"
        >
          3
        </button>
        <button
          class="btn orange"
          type="button"
        >
          +
        </button>
      </div>
      <div
        class="container-rows"
      >
        <button
          class="btn  grey wide"
          type="button"
        >
          0
        </button>
        <button
          class="btn  grey"
          type="button"
        >
          .
        </button>
        <button
          class="btn orange"
          type="button"
        >
          =
        </button>
      </div>
    </section>
  </div>
</div>
`;
